File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size()!=t.size())
            return false;
        bool flag =true;
        int i=0,j=0;
        while(flag==true and i<s.size())
        {
            flag=false;
            for(int j=0;j<t.size();j++)
            {
                if(t[j]==s[i])
                {
                    flag=true;
                    t.erase(j,1);
                    break;
                }
            }
            i++;
        }
        return flag;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s)!=len(t): return False
        maps={}
        for i in s:
            if i not in maps:
                maps[i]=0
            maps[i]+=1

        for i in t:
            if i not in maps:
                return False
            maps[i]-=1
            if maps[i]==0:
                del maps[i]
        return len(maps)==0